src/redux/actions/planActions.js

```diff
@@ -0,0 +1,329 @@
+    isFetching: false,
+    isSynced: false
+  };
+}
+
+function addPlanSuccess() {
+  return {
+    type: ADD_PLAN_SUCCESS,
+    lastUpdated: Date.now(),
+    isFetching: false,
+    isSynced: true
+  };
+}
+
+function updatePlanRequest() {
+  return {
+    type: UPDATE_PLAN_REQUEST,
+    lastUpdated: Date.now(),
+    isFetching: true,
+    isSynced: false
+  };
+}
+
+function updatePlanFailure() {
+  return {
+    type: UPDATE_PLAN_FAILURE,
+    lastUpdated: Date.now(),
+    isFetching: false,
+    isSynced: false
+  };
+}
+
+function updatePlanSuccess(plans) {
+  return {
+    type: UPDATE_PLAN_SUCCESS,
+    lastUpdated: Date.now(),
+    isFetching: false,
+    isSynced: true,
+    plans: plans
+  };
+}
+
+function deletePlanRequest() {
+  return {
+    type: DELETE_PLAN_REQUEST,
+    lastUpdated: Date.now(),
+    isFetching: true,
+    isSynced: false
+  };
+}
+
+function deletePlanFailure() {
+  return {
+    type: DELETE_PLAN_FAILURE,
+    lastUpdated: Date.now(),
+    isFetching: false,
+    isSynced: false
+  };
+}
+
+function deletePlanSuccess(plans) {
+  return {
+    type: DELETE_PLAN_SUCCESS,
+    lastUpdated: Date.now(),
+    isFetching: false,
+    isSynced: true,
+    plans: plans
+  };
+}
+
+function getPlanInfoRequest() {
+  return {
+    type: GET_PLAN_INFO_REQUEST,
+    lastUpdated: Date.now(),
+    isFetching: true,
+    isSynced: false
+  };
+}
+
+function getPlanInfoFailure() {
+  return {
+    type: GET_PLAN_INFO_FAILURE,
+    lastUpdated: Date.now(),
+    isFetching: false,
+    isSynced: false
+  };
+}
+
+function getPlanInfoSuccess(plans) {
+  return {
+    type: GET_PLAN_INFO_SUCCESS,
+    lastUpdated: Date.now(),
+    isFetching: false,
+    isSynced: true,
+    plans: plans
+  };
+}
+
+/**************************** Logical functions ****************************/
+
+/**
+ * Performs an http POST get plan info request to server.
+ * Dispatches getUserInfoRequest to indicate the beginning of a getPlanInfo process.
+ * Dispatches getPlanInfoFailure to indicate the end of a failed getPlanInfo process.
+ * Dispatches getPlanSuccess to indicate the end of a successful getPlanInfo process.
+ * If getPlanInfo process succeeds, a plan list object is received and passed into
+ * getPlanInfoSuccess to be stored into state.
+ * @returns {function(*): Promise<Response | never>} dispatch results.
+ */
+function getPlanInfo(e_id) {
+  return (dispatch, getState) => {
+    dispatch(getPlanInfoRequest());
+    return fetch('http://localhost:4201/plan/get', {
+      headers: {
+        'Content-Type': 'application/json',
+        'x-access-token': localStorage.getItem('token')
+      },
+      method: 'post',
+      body: JSON.stringify({ e_id: e_id })
+    })
+      .then(response => response.json())
+      .then(json => {
+        if (json.length === 0) dispatch(getPlanInfoFailure());
+        else {
+          // Push newly added plan to state plan list
+          let plans = getState().masterReducer.plans.plans;
```

Comment:
I might be wrong with this one. But I'm under the impression that actions shouldn't be editing the state; that's the reducer's job. Should this be sending out an action, and have the reducer deal with it? Something like dispatch(getPlanSuccess(plan))

Reply:
Your understanding is correct, but I don't believe this is editing the state. What I'm trying to accomplish here:

- make a copy of this subsection "plan" inside the state.
- change that copy.
- assign that copy to be the new subsection "plan" inside the state.